metricbeat/helper/server/udp/udp.go

```diff
@@ -43,22 +45,25 @@ func NewUdpServer(base mb.BaseMetricSet) (server.Server, error) {
 		return nil, err
 	}
 
-	listener, err := net.ListenUDP("udp", addr)
-	if err != nil {
-		return nil, err
-	}
-
-	logp.Info("Started listening for UDP on: %s:%d", config.Host, config.Port)
 	return &UdpServer{
-		listener:          listener,
+		udpaddr:           addr,
 		receiveBufferSize: config.ReceiveBufferSize,
 		done:              make(chan struct{}),
 		eventQueue:        make(chan server.Event),
 	}, nil
 }
 
-func (g *UdpServer) Start() {
+func (g *UdpServer) Start() error {
+	listener, err := net.ListenUDP("udp", g.udpaddr)
+	if err != nil {
+		return errors.Wrap(err, "Failed to start UDP server")
```

Comment:
Error strings should not be capitalized. https://github.com/golang/go/wiki/CodeReviewComments#error-strings